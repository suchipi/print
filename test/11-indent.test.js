const { print } = require("./helpers");

describe("Indentation", () => {
	const value = { a: { b: { c: { d: {} } } } };

	it("defaults to tab", () => {
		expect(print(value)).toMatchSnapshot();
	});

	it("can be overridden", () => {
		expect(print(value, { indent: "  " })).toMatchSnapshot();
		expect(print(value, { indent: " " })).toMatchSnapshot();
		expect(print(value, { indent: "" })).toMatchSnapshot();
		expect(print(value, { indent: "_hi" })).toMatchSnapshot();
	});
});
