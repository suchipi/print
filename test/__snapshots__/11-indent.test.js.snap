// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indentation can be overridden 1`] = `
{
  a: {
    b: {
      c: {
        d: {}
      }
    }
  }
}
`;

exports[`Indentation can be overridden 2`] = `
{
 a: {
  b: {
   c: {
    d: {}
   }
  }
 }
}
`;

exports[`Indentation can be overridden 3`] = `
{
a: {
b: {
c: {
d: {}
}
}
}
}
`;

exports[`Indentation can be overridden 4`] = `
{
_hia: {
_hi_hib: {
_hi_hi_hic: {
_hi_hi_hi_hid: {}
_hi_hi_hi}
_hi_hi}
_hi}
}
`;

exports[`Indentation defaults to tab 1`] = `
{
	a: {
		b: {
			c: {
				d: {}
			}
		}
	}
}
`;
